packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -155,13 +155,7 @@ export function useGridDimensions(
     ) {
       apiRef.current.publishEvent('viewportInnerSizeChange', newFullDimensions.viewportInnerSize);
     }
-  }, [
-    apiRef,
-    props.scrollbarSize,
-    props.autoHeight,
-    rowsMeta.currentPageTotalHeight,
-    totalHeaderHeight,
-  ]);
+  }, [apiRef, props.scrollbarSize, props.autoHeight, totalHeaderHeight]);
```

Comment:
Alright, sounds good to me!
This PR might be a potential fix for the same issue on the `master` branch since we won't change the overlays layout in v5.